### PR TITLE
Add method hasBuildDependency to BuildVerifier

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy [hunk 1]
```diff
@@ -26,4 +26,6 @@ interface BuildTestVerifier {
     boolean hasDependency(String artifactId)
 
     boolean hasTestResourceDependencyWithGroupId(String expectedGroupId)
+
+    boolean hasBuildDependency(String id)
```
Comment (timyates) [inline starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy]: `hasBuildPlugin`?

Comment (timyates) [reply starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy]: Also, where is this used?  Do we need to implement it for Maven?

**Review by timyates — APPROVED** [review]
Approved, but I am not sure where this is used...

File: starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy]: I use it in tests in upcoming PRs

Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/build/BuildTestVerifier.groovy]: renamed: 

https://github.com/micronaut-projects/micronaut-starter/pull/1662/commits/d4cb18cfa1567c9651c5d21ddcd0aa824b969ba9

Test using it: 
https://github.com/micronaut-projects/micronaut-starter/pull/1662/commits/76318cae290d489bc85d4721367a855a7fcdd790

**Comment by timyates** [comment]
image

image

😅

**Pull request merged**